subsys/net/lib/sockets/sockets_packet.c

```diff
@@ -360,10 +360,16 @@ static int packet_sock_setsockopt_vmeth(void *obj, int level, int optname,
 	return zpacket_setsockopt_ctx(obj, level, optname, optval, optlen);
 }
 
+static int packet_sock_close_vmeth(void *obj)
+{
+	return zsock_close_ctx(obj);
```

Comment:
Are all operations done by `zsock_close_ctx()` beside `net_context_put()` (callback reset, packet flushing) also valdi for packet socket? It seems ok to use it, but I wanted to confirm that, other APIs in this implementation operate directly on net_context layer.

Reply:
That is a good question. I thought about that too, but did not spot anything that needs to be done differently in packet socket, so easiest is to call the same function as for normal sockets.

Reply:
I checked that too, and as far as I can tell, it follows architecture of the TCP/UDP sockets, so reusing its close method should be fine.